packages/big-design/src/components/Select/Select.tsx

```diff
@@ -263,95 +190,130 @@ export class Select extends React.PureComponent<SelectProps, SelectState> {
-
-          return React.cloneElement(child, {
-            'data-highlighted': this.state.highlightedItem && actionId === this.state.highlightedItem.id,
-            id: actionId,
-            onClick: this.handleOnActionClick,
-            onMouseOver: this.handleOnItemMouseOver,
-            ref,
-            role: 'option',
-          }) as React.LiHTMLAttributes<HTMLLIElement>;
-        default:
-          return;
+        return multi ? (
+          <ListCheckboxItem
+            {...rest}
+            aria-selected={this.isChecked(option)}
+            checked={this.isChecked(option)}
+            data-highlighted={isHighlighted}
+            id={id}
+            key={index}
+            onClick={event => {
+              event.stopPropagation();
+              event.preventDefault();
+              this.handleOnCheckboxOptionClick(option);
+            }}
+            onFocus={this.handleOnOptionHighlighted}
+            onMouseOver={this.handleOnOptionHighlighted}
+            ref={ref}
+            role="option"
+          >
+            {content}
+          </ListCheckboxItem>
+        ) : (
+          <ListItem
+            {...rest}
+            {...ariaSelected}
+            actionType="normal"
+            data-highlighted={isHighlighted}
+            id={id}
+            key={index}
+            onClick={() => this.handleOnOptionClick(option)}
+            onFocus={this.handleOnOptionHighlighted}
+            onMouseOver={this.handleOnOptionHighlighted}
+            ref={ref}
+            role="option"
+          >
+            {content}
+          </ListItem>
+        );
       }
     });
   }
 
+  private renderAction() {
+    const { action } = this.props;
+    const { highlightedItem } = this.state;
+
+    if (!action || !action.content) {
+      return null;
+    }
+
+    const id = this.getActionId(action);
+    const isHighlighted = Boolean(highlightedItem && id === highlightedItem.id);
+    const ref = React.createRef<HTMLLIElement>();
+
+    if (!action.disabled) {
+      this.listItems.push({ item: action, ref });
+    }
+
+    const { content, icon, ...rest } = action;
+
+    return (
+      <Box borderTop="box" marginTop="xSmall" paddingTop="xSmall">
+        <ListItem
+          {...rest}
+          data-highlighted={isHighlighted}
+          id={id}
+          onClick={() => this.handleOnActionClick(action)}
+          onFocus={this.handleOnOptionHighlighted}
+          onMouseOver={this.handleOnOptionHighlighted}
+          ref={ref}
+          role="option"
+        >
+          <Flex alignItems="center" flexDirection="row">
+            {icon && <FlexItem paddingRight="xSmall">{this.renderIcon(action, isHighlighted)}</FlexItem>}
+            {content}
+          </Flex>
+        </ListItem>
+      </Box>
+    );
+  }
+
+  private renderIcon(action: Action, isHighlighted: boolean) {
+    return (
+      React.isValidElement(action.icon) &&
+      React.cloneElement(action.icon, {
+        color: action.disabled
```

Comment:
🍹Could you refactor these ternary statements into something a little more legible?

Reply:
Tried to make it more legible. Let me know if you have a better idea.

Reply:
I was thinking of something along the lines of what @deini put below.